api/v1alpha1/extension_types.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type ExtensionManagedState string
+
+const (
+	// Peform reconcilliation of this Extension
+	ManagedStateActive ExtensionManagedState = "Active"
+	// Pause reconcilliation of this Extension
+	ManagedStatePaused ExtensionManagedState = "Paused"
+)
+
+type ExtensionSourcePackage struct {
+	//+kubebuilder:validation:MaxLength:=48
```

Comment:
Why 48? 

I took a look at the operatorhubio catalog and some of the package names get to the high thirties with the highest I saw being a length 40 which makes me feel that putting an arbitrary limit to package name length being 48 might be insufficient. I'm also not aware of this length being enforced during the catalog validation process, but I could just not be aware of it.

For reference, this is the command I ran to check the operatorhubio catalog package name lengths:
```sh
opm render quay.io/operatorhubio/catalog:latest | jq -s '.[] | select(.schema == "olm.package") | .name | length'
```

Reply:
48 is what the ClusterExtension API uses for these fields.
https://github.com/operator-framework/operator-controller/blob/2df4bceb4cbab606a5e9dd14490900a499d27354/api/v1alpha1/clusterextension_types.go#L43-L45

Reply:
It is because Rukpak reserves some characters for its use so that it can use the full clusterextension name when it creates secondary objects for the clusterextension.

As much as possible, it is best to fully define, enforce, and err towards overly strict schemas because once we GA, we cannot relax them without a breaking change.